plugin/src/main/kotlin/nl/avisi/anowalke/NlpStrategy.kt

```diff
@@ -0,0 +1,99 @@
+        return identifiers
+    }
+
+    fun searchFilesRecusively(file: VirtualFile, project: Project): List<Identifier>? {
+        log.warn("Initiate getFilesAsIdentifier function.")
+        // Zoek naar files in de java map ipv files met filetype java, omdat hij sommige filetypes door elkaar haalt
+        return if (file.isDirectory && file.name == "java") {
+            log.warn("Current file: '${file.name}' is a java directory.")
+            val identifiers = getIdentifiersFromFile(file, project)
+            log.warn("getJavaFiles function retrieved ${identifiers?.size} identifiers.")
+            identifiers
+        } else if (file.isDirectory && file.name != "java") {
+            log.warn("Current file: ${file.name} is not a java directory. Check Children for a directory named 'java'.")
+            file.children.flatMap { searchFilesRecusively(it, project).orEmpty() }
+        } else {
+            log.warn("Current file: ${file.name} is not a directory. Ignore file.")
+            null
+        }
+    }
+
+    private fun getIdentifiersFromFile(file: VirtualFile, project: Project): List<Identifier>? {
+        log.warn("Initiate getJavaFiles function.")
+        // Filter out hidden files
+        if (!file.isDirectory && !file.name.startsWith(".")) {
+            log.warn("File: ${file.name} is not a directory nor a hidden file.")
+            val expressions = PsiManager.getInstance(project).findFile(file)?.let { psiFile ->
+                psiService.extract(psiFile)
+            }
+            if (expressions != null) {
+                log.warn("Extract function retrieved ${expressions.size} expressions.")
+                return sanitiser
+                    .sanitise(expressions)
+                    .map { expression -> Identifier(file.name, expression) }
+            }
+        } else if (file.isDirectory) {
+            log.warn("File: ${file.name} is a directory. Check children for java files.")
+            return file.children.flatMap { getIdentifiersFromFile(it, project).orEmpty() }
+        }
+        return null
+    }
+
+    private fun getCodeSmellLocations(identifiers: List<Identifier>, codeSmell: String): List<KeywordLocation> {
+        var processedIdentifiers = identifiers
+            .groupBy { it.expression }
+        when (codeSmell) {
+            "Shotgun Surgery" -> processedIdentifiers = processedIdentifiers
+                .filter {
+                    it.value.size > 1
+                }
+        }
+        val listOfKeywordLocations = processedIdentifiers.map { mapOfKeywordAndIdentifier ->
+            val keywordLocation = KeywordLocation()
+            keywordLocation.keywoard = mapOfKeywordAndIdentifier.key
+            keywordLocation.classesContainingKeyword = mapOfKeywordAndIdentifier.value
+                .map { identifier ->
+                    identifier.className
+                }
+            keywordLocation
+        }
```

Comment:
mapOfKeywordAndIdentifierkan je hier ook al exploden als het goed is. Dus dan krijg je het volgende (ook nog kleine verbetering gedaan voor het teruggeven van KeywordLocation. Zou het wel een dataclass moeten zijn. of iig die constructor moeten hebben

```suggestion
        val listOfKeywordLocations = processedIdentifiers.map { (keyword, identifiers) ->
            KeywordLocation(
                keyword = keyword,
                classesContainingKeyword = identifiers.map { identifier ->
                    identifier.className
                }
            )
        }
```

Reply:
Ik geloof dat ik dit inmiddels al gedaan heb? Of niet?